### PR TITLE
[web-browser][web] Create state before open popup

### PR DESCRIPTION
**Pull request by sreuter**
Based on https://github.com/expo/expo/issues/10459#issuecomment-720446251 suggestion, this prevents Desktop Safari from blocking opening the popup, as the async call doesn't delay the `window.open` call anymore.

# Why

Opening a popup via `promptAsync`does not work on Desktop Safari, as creating the `state` delays the window from opening for too long. Moving the state generation to after the window is opened fixes this.

# How

Move state generation to after the popup is opened.

# Test Plan

Use `promptAsync` in Desktop Safari without an existing `state` on the `authorizationUrl`.  With the change applied, the popup should not be blocked anymore by Desktop Safari.

# Checklist

- [x ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x ] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [x ] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Review by bbarthec — COMMENTED** [review]
Looking good 👍 
Please add [CHANGELOG entry](https://github.com/expo/expo/blob/main/CONTRIBUTING.md#-before-submitting) and we're ready to merge it 😊

**Comment by sreuter** [comment]
> Looking good 👍
> Please add [CHANGELOG entry](https://github.com/expo/expo/blob/main/CONTRIBUTING.md#-before-submitting) and we're ready to merge it 😊

Cheers @bbarthec, should be all good to go now!

File: packages/expo-web-browser/CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,8 @@
 
 ### 🐛 Bug fixes
 
+- Fix popup being blocked by Safari. ([#17222](https://github.com/expo/expo/pull/17222) by [@sreuter](https://github.com/sreuter))
```
Comment (bbarthec) [inline packages/expo-web-browser/CHANGELOG.md:11]:
```suggestion
- On Web fix popup being blocked by Safari. ([#17222](https://github.com/expo/expo/pull/17222) by [@sreuter](https://github.com/sreuter))
```

**Review by bbarthec — APPROVED** [review]
Looking good 👍 
Thank you 😁

**Comment by sreuter** [comment]
@bbarthec Any chance we could publish a new version with the latest fixes?

**Comment by sreuter** [comment]
Sweet stuff! Pulled into our project and removed our current workaround (e.g. generating the state on our own).. Thanks for helping getting this through @bbarthec, you're amazing! 🙌

**Pull request merged**